Projects/Server/Buffers/CircularBufferReader.cs

```diff
@@ -239,14 +235,14 @@ public uint ReadUInt32()
                             .IndexOf(default(T)) * sizeT;
 
                         int secondLength = index < 0 ? remaining : index;
-                        int length = First.Length + secondLength;
+                        int length = firstLength + secondLength;
 
                         // Assume no strings should be too long for the stack
                         Span<byte> bytes = stackalloc byte[length];
                         First.Slice(Position).CopyTo(bytes);
-                        Second.Slice(0, secondLength).CopyTo(bytes.Slice(First.Length));
+                        Second.Slice(0, secondLength).CopyTo(bytes.Slice(firstLength));
 
-                        Position += length;
+                        Position += length + (index >= 0 ? sizeT : 0);
```

Comment:
We can't just add 1 or 2 for the null value. We need to actively zero it out.
Should be changed to:
`Write((byte)0);` and `Write((ushort)0);` respectively.